diff --git a/src/Controller/UserActionController.php b/src/Controller/UserActionController.php
index f939cb8..524a343 100644
--- a/src/Controller/UserActionController.php
+++ b/src/Controller/UserActionController.php
@@ -90,10 +90,7 @@ class UserActionController implements ContainerInjectionInterface {
     // or a simple page to logged in users explaining their state.
     if (is_null($service)) {
       if (!$this->userHasSingleSignOnSession(NULL)) {
-        return $this->formBuilder->getForm(
-          '\Drupal\cas_server\Form\UserLogin',
-          ''
-        );
+        return \Drupal\Core\Access\AccessResult::forbidden();
       }
       else {
         return $this->generateLoggedInMessage();
