diff --git a/src/Controller/UserActionController.php b/src/Controller/UserActionController.php
index f939cb8..4539a6a 100644
--- a/src/Controller/UserActionController.php
+++ b/src/Controller/UserActionController.php
@@ -63,6 +63,11 @@ class UserActionController implements ContainerInjectionInterface {
    */
   public function login(): TrustedRedirectResponse|array {
     $this->killSwitch->trigger();
+
+    if($this->currentUser->isAuthenticated()) {
+      return $this->generateNotPermittedMessage();
+    }
+
     $request = $this->requestStack->getCurrentRequest();
     $service = $request->query->has('service') ? $request->query->get('service') : NULL;
     $service_entity = $this->configHelper->loadServiceFromUri($service);
