diff --git a/src/Controller/UserActionController.php b/src/Controller/UserActionController.php
index f939cb8..840f1e5 100644
--- a/src/Controller/UserActionController.php
+++ b/src/Controller/UserActionController.php
@@ -135,6 +135,10 @@ class UserActionController implements ContainerInjectionInterface {
       return $response;
     }
 
+    if(empty($this->currentUser)) {
+      return \Drupal\Core\Access\AccessResult::forbidden();
+    }
+
     // Present the user with a login form.
     return $this->formBuilder->getForm(
       '\Drupal\cas_server\Form\UserLogin',
