diff --git a/core/includes/install.core.inc b/core/includes/install.core.inc
index 00eadeb5ec4..177baab458c 100644
--- a/core/includes/install.core.inc
+++ b/core/includes/install.core.inc
@@ -1891,11 +1891,6 @@ function install_finished(&$install_state) {
   // installer does not use an HttpKernel, that event is never triggered.
   \Drupal::service('router.builder')->rebuild();
 
-  // Run cron to populate update status tables (if available) so that users
-  // will be warned if they've installed an out of date Drupal version.
-  // Will also trigger indexing of profile-supplied content or feeds.
-  \Drupal::service('cron')->run();
-
   if ($install_state['interactive']) {
     // Load current user and perform final login tasks.
     // This has to be done after drupal_flush_all_caches()
