diff --git a/src/Plugin/EntityPrint/PrintEngine/DomPdf.php b/src/Plugin/EntityPrint/PrintEngine/DomPdf.php
index a6d6db8..6b71ffe 100644
--- a/src/Plugin/EntityPrint/PrintEngine/DomPdf.php
+++ b/src/Plugin/EntityPrint/PrintEngine/DomPdf.php
@@ -236,11 +236,17 @@ class DomPdf extends PdfEngineBase implements ContainerFactoryPluginInterface {
    */
   protected function doRender() {
     $this->setupHttpContext();
+    $timestamp = date('U').rand(111,999);
+    $str = $this->dompdf->getDom()->C14N();
 
     if (!$this->hasRendered) {
       try {
         $this->dompdf->render();
         $this->hasRendered = TRUE;
+        if(!empty($str) && (str_contains($str,'webform-element') || str_contains($str,'webform-codemirror-runmode'))) {
+          $font = $this->dompdf->getFontMetrics()->getFont("helvetica", "regular");
+          $this->dompdf->getCanvas()->page_text(6, 6, "DOCUMENT ID: $timestamp - Page {PAGE_NUM} of {PAGE_COUNT}", $font, 6, array(0,0,0));
+        }
       }
       catch (DompdfLibException $e) {
         throw new PrintEngineException(sprintf('Failed to generate PDF: %s', $e));
