diff --git a/src/Plugin/Block/LBKBaseComponent.php b/src/Plugin/Block/LBKBaseComponent.php
index e52fe63..6d87ebf 100644
--- a/src/Plugin/Block/LBKBaseComponent.php
+++ b/src/Plugin/Block/LBKBaseComponent.php
@@ -46,6 +46,13 @@ abstract class LBKBaseComponent extends BlockBase
   use PluginWithFormsTrait;
   use StringTranslationTrait;
 
+  /**
+   * The base ID for the component.
+   *
+   * @var string
+   */
+  protected $baseId;
+
   /**
    * The transliteration service.
    *
@@ -82,6 +89,8 @@ abstract class LBKBaseComponent extends BlockBase
    *   The EntityTypeBundleInfo service.
    */
   public function __construct(array $configuration, $plugin_id, $plugin_definition, CurrentRouteMatch $currentRouteMatch, EntityTypeBundleInfo $entityTypeBundleInfo) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+
     $this->setConfiguration($configuration);
     $this->currentRouteMatch = $currentRouteMatch;
     $this->entityTypeBundleInfo = $entityTypeBundleInfo;
diff --git a/src/Plugin/Block/LBKTab/LBKTab.php b/src/Plugin/Block/LBKTab/LBKTab.php
index 6954f5c..faef4a8 100755
--- a/src/Plugin/Block/LBKTab/LBKTab.php
+++ b/src/Plugin/Block/LBKTab/LBKTab.php
@@ -190,7 +190,7 @@ class LBKTab extends LBKBaseComponent implements ContainerFactoryPluginInterface
         $tab_text = '';
         if (isset($tabs[$i]['text_format']['value'])) $tab_text = $tabs[$i]['text_format']['value'];
 
-        $tab_text_format = '';
+        $tab_text_format = 'full_html';
         if (isset($tabs[$i]['text_format']['format'])) $tab_text_format = $tabs[$i]['text_format']['format'];
 
         $form['items_fieldset']['items'][$i]['text_format'] = [
@@ -261,6 +261,7 @@ class LBKTab extends LBKBaseComponent implements ContainerFactoryPluginInterface
         'wrapper' => 'items-fieldset-wrapper',
       ],
       '#weight' => 10,
+      '#access' => FALSE,
       '#suffix' => '</div>',
     ];
 
