diff --git a/src/Plugin/Mail/MailgunMail.php b/src/Plugin/Mail/MailgunMail.php
index f703160b4..f46618697 100644
--- a/web/modules/contrib/mailgun/src/Plugin/Mail/MailgunMail.php
+++ b/web/modules/contrib/mailgun/src/Plugin/Mail/MailgunMail.php
@@ -234,6 +234,9 @@ protected function buildMessage(array $message) {
       if (substr($key, 0, 2) == 'h:' || substr($key, 0, 2) == 'v:') {
         $mailgun_message[$key] = $value;
       }
+      if($key == 'recipient-variables') {
+        $mailgun_message[$key] = json_encode($value);
+      }
     }
 
     // Mailgun will accept the message but will not send it.
