diff --git a/src/Plugin/Mail/MailgunMail.php b/src/Plugin/Mail/MailgunMail.php
index f703160b4..a653f28aa 100644
--- a/src/Plugin/Mail/MailgunMail.php
+++ b/src/Plugin/Mail/MailgunMail.php
@@ -231,9 +231,12 @@ protected function buildMessage(array $message) {
         $mailgun_message[$key] = $value;
       }
       // Check for custom MIME headers or custom JSON data.
-      if (substr($key, 0, 2) == 'h:' || substr($key, 0, 2) == 'v:') {
+      if (substr($key, 0, 2) == 't:' || substr($key, 0, 2) == 'h:' || substr($key, 0, 2) == 'v:' || $key == 'recipient-variables' || $key == 'template') {
         $mailgun_message[$key] = $value;
       }
+      if($key == 'template') {
+        unset($mailgun_message['html']);
+      }
     }
 
     // Mailgun will accept the message but will not send it.
