diff -ur pantheon_secrets/src/Plugin/KeyProvider/PantheonSecretKeyProvider.php /home/jfoust/Lando/webpresence-two/web/modules/contrib/pantheon_secrets/src/Plugin/KeyProvider/PantheonSecretKeyProvider.php
--- pantheon_secrets/src/Plugin/KeyProvider/PantheonSecretKeyProvider.php	2023-12-15 09:37:56.928521483 -0500
+++ pantheon_secrets/src/Plugin/KeyProvider/PantheonSecretKeyProvider.php	2023-12-15 09:32:58.766794376 -0500
@@ -47,6 +47,7 @@
   public function defaultConfiguration() {
     return [
       'secret_name' => '',
+      'base64_encoded' => FALSE,
     ];
   }
 
@@ -73,7 +74,17 @@
       ];
     }
 
-    return $form;
+    // If this key type is for an encryption key.
+    if ($form_state->getFormObject()->getEntity()->getKeyType()->getPluginDefinition()['group'] == 'encryption') {
+      // Add an option to indicate that the value is stored Base64-encoded.
+      $form['base64_encoded'] = [
+        '#type' => 'checkbox',
+        '#title' => $this->t('Base64-encoded'),
+        '#description' => $this->t('Checking this will store the key with Base64 encoding.'),
+        '#default_value' => isset($this->getConfiguration()['base64_encoded']) ? $this->getConfiguration()['base64_encoded'] : $this->defaultConfiguration()['base64_encoded'],
+      ];
+    }
+      return $form;
   }
 
   /**
@@ -143,9 +154,15 @@
       return NULL;
     }
 
-    return $secret->getValue();
+    $key_value = $secret->getValue();
 
-  }
+    if (isset($this->configuration['base64_encoded']) && $this->configuration['base64_encoded'] == TRUE) {
+      $key_value = base64_decode($key_value);
+    }
+
+      return $key_value;
+
+    }
 
   /**
    * {@inheritdoc}
