diff --git a/site_audit_report_entity/src/SiteAuditReportListBuilder.php b/site_audit_report_entity/src/SiteAuditReportListBuilder.php
index 4ff425e..ec7bb43 100644
--- a/site_audit_report_entity/src/SiteAuditReportListBuilder.php
+++ b/site_audit_report_entity/src/SiteAuditReportListBuilder.php
@@ -50,6 +50,7 @@ class SiteAuditReportListBuilder extends EntityListBuilder {
     // Sorting my entities.
     $entity_query = \Drupal::entityQuery('site_audit_report')
       ->sort('created' , 'DESC');
+    $entity_query->accessCheck(TRUE);
 
     // Make the table sortable.
     $header = $this->buildHeader();
diff --git a/site_audit_send/src/RestClient.php b/site_audit_send/src/RestClient.php
index 158d9fd..46f4b73 100644
--- a/site_audit_send/src/RestClient.php
+++ b/site_audit_send/src/RestClient.php
@@ -19,6 +19,13 @@ use Symfony\Component\DependencyInjection\ContainerInterface;
  */
 class RestClient {
 
+  /**
+   * The config factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
   /**
    * Constructs a \Drupal\system\ConfigFormBase object.
    *
