diff --git a/components/cards/paragraph--uswds-card-group-flag.html.twig b/components/cards/paragraph--uswds-card-group-flag.html.twig
index 9f62650..46013a9 100644
--- a/components/cards/paragraph--uswds-card-group-flag.html.twig
+++ b/components/cards/paragraph--uswds-card-group-flag.html.twig
@@ -39,12 +39,7 @@
 
 {# The template default set classes. #}
 {# Sets Paragraph ID as class. #}
-{% set classes = [
-  'paragraph',
-  'paragraph--type--' ~ paragraph.bundle|clean_class,
-  view_mode ? 'paragraph--view-mode--' ~ view_mode|clean_class,
-  'paragraph--id--' ~ paragraph.id.value,
-] %}
+{% set classes = [] %}
 
 {{ attach_library('uswds_paragraph_components/uswds-cards') }}
 
diff --git a/components/cards/paragraph--uswds-card-group-regular.html.twig b/components/cards/paragraph--uswds-card-group-regular.html.twig
index 97c8a44..6ab8d26 100644
--- a/components/cards/paragraph--uswds-card-group-regular.html.twig
+++ b/components/cards/paragraph--uswds-card-group-regular.html.twig
@@ -39,12 +39,7 @@
 
 {# The template default set classes. #}
 {# Sets Paragraph ID as class. #}
-{% set classes = [
-  'paragraph',
-  'paragraph--type--' ~ paragraph.bundle|clean_class,
-  view_mode ? 'paragraph--view-mode--' ~ view_mode|clean_class,
-  'paragraph--id--' ~ paragraph.id.value,
-] %}
+{% set classes = [] %}
 
 {{ attach_library('uswds_paragraph_components/uswds-cards') }}
 
