diff --git a/src/Plugin/WebformElement/WebformSignature.php b/src/Plugin/WebformElement/WebformSignature.php
index 9536d12cd..350b31efc 100644
--- a/src/Plugin/WebformElement/WebformSignature.php
+++ b/src/Plugin/WebformElement/WebformSignature.php
@@ -53,7 +53,7 @@ class WebformSignature extends WebformElementBase implements WebformElementFileD
       // General settings.
       'description' => $this->t('Sign above'),
       'readonly' => FALSE,
-      'uri_scheme' => 'public',
+      'uri_scheme' => 'private',
     ] + parent::defineDefaultProperties();
     unset(
       $properties['format_items'],
@@ -346,7 +346,11 @@ class WebformSignature extends WebformElementBase implements WebformElementFileD
   public static function getVisibleStreamWrappers() {
     /** @var \Drupal\Core\StreamWrapper\StreamWrapperManagerInterface $stream_wrapper_manager */
     $stream_wrapper_manager = \Drupal::service('stream_wrapper_manager');
-    return $stream_wrapper_manager->getNames(StreamWrapperInterface::WRITE_VISIBLE);
+    $stream_wrappers = $stream_wrapper_manager->getNames(StreamWrapperInterface::WRITE_VISIBLE);
+    if (!\Drupal::config('webform.settings')->get('file.file_public')) {
+      unset($stream_wrappers['public']);
+    }
+    return $stream_wrappers;
   }
 
   /**
